pymeasure/experiment/procedure.py

```diff
@@ -180,6 +182,38 @@ def set_parameters(self, parameters, except_missing=True):
                     raise NameError("Parameter '{}' does not belong to '{}'".format(
                         name, repr(self)))
 
+    def _update_metadata(self):
+        """ Collects all the Metadata objects for the procedure and stores
+        them in a meta dictionary so that the actual values can be set and used
+        in their stead
+        """
+        if not self._metadata:
+            self._metadata = {}
+
```

Comment:
I am not sure why this check is needed. Is it to copy class attribute `_metadata` at instance level ?
I think internal method `_update_metadata` is called only once at `__init__`, so why not to replace the code with simply: `self._metadata = {}` avoiding the `if` and removing line 72.

Reply:
I think that this if statement is indeed not necessary and can be replaced by `self._metadata = {}`. I don't completely understand what you mean by removing line 72 though??

Reply:
Sorry I meant line 67 where dictionary `_metadata` is declared as class attribute.

Reply:
Hmm ahh yeah that makes sense. Thank!